product_pricelist_supplierinfo/models/product_template.py

```diff
@@ -13,40 +13,12 @@ def _get_supplierinfo_pricelist_price(
             self, rule, date=None, quantity=None, product_id=None):
         """Method for getting the price from supplier info."""
         self.ensure_one()
+        product = self.product_variant_id
         if product_id:
-            domain = [
-                '|',
-                ('product_id', '=', product_id),
-                ('product_tmpl_id', '=', self.id),
-            ]
-        else:
-            domain = [
-                ('product_tmpl_id', '=', self.id),
-            ]
```

Comment:
I think that this part of domain can be replaced by 
                "|",
                ("product_id", "=", self.id),
                "&",
                ("product_tmpl_id", "=", self.product_tmpl_id.id),
                ("product_id", "=", False),

Reply:
@sergio-teruel we are removing all that domain stuff and using `_select_seller` instead.

Reply:
yeahhh I saw!! I have done a comment